storage/local/persistence.go

```diff
@@ -606,7 +608,9 @@ func (p *persistence) checkpointSeriesMapAndHeads(fingerprintToSeries *seriesMap
 			if err != nil {
 				return
 			}
-			w.Write(buf)
+			if _, err = w.Write(buf); err != nil {
```

Comment:
@fabxc How would it still work? We shadow `err` at the beginning of the function and then continue using the new `err` instead of the named return value `err`. Combined with naked returns, how would the return value ever get properly set?